sphinx/builders/html/transforms.py

```diff
@@ -47,7 +47,7 @@ def run(self, **kwargs: Any) -> None:
         matcher = NodeMatcher(nodes.literal, classes=["kbd"])
         # this list must be pre-created as during iteration new nodes
         # are added which match the condition in the NodeMatcher.
-        for node in list(self.document.findall(matcher)):  # type: nodes.literal
+        for node in list(matcher.find_all_in(self.document)):
```

Comment:
i'm baffled as to why this needs to be wrapped in `list(...)` but it seems to break without it

???